core/controllers/profile.py

```diff
@@ -664,22 +799,40 @@ def get(self):
             })
 
     @acl_decorators.open_access
-    def put(self):
+    def put(self) -> None:
         """Handles PUT requests."""
-        user_has_viewed_lesson_info_modal_once = self.normalized_payload.get(
-            'user_has_viewed_lesson_info_modal_once')
+        # In frontend, we are calling this put method iff the user is
+        # logged-in, so here we sure that self.user_id is never going
+        # to be None, but to narrow down the type and handle the None
+        # case gracefully, we are returning if self.user_id is None.
+        if self.user_id is None:
```

Comment:
Basically, this method is given open_access by applying the `open_access` decorator. So we just can't use assert here, and here we are returning early if self.user_id is None because we are sure that from the front end we are never calling this handler for a not-logged user. You can refer to this conversation https://github.com/oppia/oppia/pull/16501#discussion_r1021066021 where this solution is concluded.

Reply:
makes sense